app/knight/knight_class.py

```diff
@@ -0,0 +1,51 @@
+class Knight:
+    protection = 0
+    list_of_knights = {}
+
+    def __init__(self, knight: object) -> None:
+        # Major values
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+
+        # Adding properties to addition features
+        self.search_addition_armor(knight["armour"])
+        self.add_power_of_weapon(knight["weapon"])
+        self.add_properties_of_potion(knight["potion"])
+
+        # Adding all knights in the class list
+        self.list_of_knights[self.name] = self
+
+    def __repr__(self) -> str:
+        return f"{self.hp}"
+
+    def search_addition_armor(self, armor: object) -> None:
+        sum_of_armor = 0
+
+        for part in armor:
+            sum_of_armor += part["protection"]
+
+        self.protection += sum_of_armor
+
+    def add_power_of_weapon(self, weapon: object) -> None:
+        self.power += weapon["power"]
+
+    def add_properties_of_potion(self, potion: object) -> None:
+        if potion is not None:
+            if "power" in potion["effect"]:
```

Comment:
Prefer using `for` loops instead of `if` statements as mentioned [in the checklist](https://github.com/mate-academy/py-knights-fighting/blob/master/checklist.md).

Reply:
+